src/main/java/com/epam/grid/engine/provider/queue/slurm/SlurmQueueProvider.java

```diff
@@ -0,0 +1,398 @@
+        checkIsResultIsCorrect(result);
+        return Queue.builder()
+                .name(registrationRequest.getName())
+                .hostList(registrationRequest.getHostList())
+                .allowedUserGroups(registrationRequest.getAllowedUserGroups())
+                .slots(new SlotsDescription(2, Collections.emptyMap()))
+                .build();
+    }
+
+    @Override
+    public Queue updateQueue(final QueueVO updateRequest) {
+        checkRegistrationRequest(updateRequest);
+
+        final String updateName = updateRequest.getName();
+        final List<String> updateHostList = updateRequest.getHostList();
+        final List<String> updateUserGroups = updateRequest.getAllowedUserGroups();
+        if (updateName == null || updateHostList == null || updateUserGroups == null) {
+            throw new UnsupportedOperationException("Name, hostList and allowedUserGroups should be specified for "
+                    + "successful partition update");
+        }
+
+        final Context context = createContextWithUpdatePartitionName(updateName);
+        final CommandResult sinfoResult = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SINFO_COMMAND, context));
+        checkIfExecutionResultIsEmpty(sinfoResult);
+
+        final List<List<String>> partitionData = getNodesData(sinfoResult.getStdOut());
+
+        final List<String> currentNodesDecrypted = getNodesFromPartitionDataAndDecrypt(partitionData);
+        final List<String> userGroups = getUserGroupsFromPartitionData(partitionData);
+        final int cpus = getCpusFromPartitionData(partitionData);
+
+        final List<String> updateHostListDecrypted = decryptGroupOfNodes(updateHostList);
+        sortList(updateUserGroups);
+
+        if (updateHostListDecrypted.equals(currentNodesDecrypted) && updateUserGroups.equals(userGroups)) {
+            throw new GridEngineException(HttpStatus.BAD_REQUEST, "New partition properties and the current one are "
+                    + "equal");
+        }
+        fillContextWithDateToUpdate(context, updateUserGroups, updateHostListDecrypted);
+
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SCONTROL_COMMAND, context));
+        checkIsResultIsCorrect(result);
+
+        return Queue.builder()
+                .name(updateName)
+                .hostList(updateHostListDecrypted)
+                .allowedUserGroups(updateUserGroups)
+                .slots(new SlotsDescription(cpus, Collections.emptyMap()))
+                .build();
+    }
+
+    @Override
+    public List<Queue> listQueues() {
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SINFO_COMMAND, new Context()));
+        checkIsResultIsCorrect(result);
+
+        if (result.getStdOut().size() == 0) {
+            return Collections.EMPTY_LIST;
```

Comment:
if we use `Collections.emptyList()` or just `List.of()` we could avoid `Unchecked assignment: 'java.util.List' to 'java.util.List<Queue>` warning here
the same on line 183

Reply:
fixed